docs/developer-guide.md

```diff
@@ -30,3 +30,28 @@ bazel clean --expunge_async
 * Don't add `#!/usr/bin/env bash` or equivalent lines to any modules/files. This line will be added during the compilation.
 * Put all code in functions. The merge logic will add the call to `main`. In other words, don't add any entry point to actually
 execute the script.
+
+
+## Writing Tests
+We integrated [`bats`](https://github.com/bats-core/bats-core) as our
+testing system. To add tests, read the
+[official documentation](https://bats-core.readthedocs.io/en/latest/)
+and update/add `*.bats` files to the `asmcli/tests` folder.
+If there are any set-ups that could be shared among tests, update
```

Comment:
Iif there is any set-up that could be shared..."

Reply:
Done